File: mobile/src/types/instructor-exercise.ts
export type InstructorExerciseSource = 'manual' | 'ai_saved';

export interface InstructorExercise {
  id: string;
  instructor_id: string;
  name_he: string;
  name_en: string | null;
  description_he: string | null;
  difficulty_level: string;
  body_focus: string[];
  possible_equipment: string[];
  teaching_notes: string | null;
  source: InstructorExerciseSource;
  based_on_catalog_exercise_id: string | null;
  based_on_modification: string | null;
  is_archived: boolean;
  created_at: string;
  updated_at: string;
}

export interface CreateInstructorExerciseInput {
  name_he: string;
  name_en?: string | null;
  description_he?: string | null;
  difficulty_level: string;
  body_focus?: string[];
  possible_equipment?: string[];
  teaching_notes?: string | null;
  source?: InstructorExerciseSource;
  based_on_catalog_exercise_id?: string | null;
  based_on_modification?: string | null;
}

export interface UpdateInstructorExerciseInput {
  name_he?: string;
  name_en?: string | null;
  description_he?: string | null;
  difficulty_level?: string;
  body_focus?: string[];
  possible_equipment?: string[];
  teaching_notes?: string | null;
  is_archived?: boolean;
}

File: backend/app/schemas/instructor_exercise.py
from typing import Literal

from pydantic import BaseModel, Field


class InstructorExerciseResponse(BaseModel):
    id: str
    instructor_id: str
    name_he: str
    name_en: str | None = None
    description_he: str | None = None
    difficulty_level: str
    body_focus: list[str]
    possible_equipment: list[str]
    teaching_notes: str | None = None
    source: str
    based_on_catalog_exercise_id: str | None = None
    based_on_modification: str | None = None
    is_archived: bool
    created_at: str
    updated_at: str


class InstructorExerciseListResponse(BaseModel):
    exercises: list[InstructorExerciseResponse]
    count: int


class CreateInstructorExerciseRequest(BaseModel):
    name_he: str = Field(min_length=1)
    name_en: str | None = None
    description_he: str | None = None
    difficulty_level: Literal["beginner", "intermediate", "advanced"]
    body_focus: list[str] = Field(default_factory=list)
    possible_equipment: list[str] = Field(default_factory=list)
    teaching_notes: str | None = None
    source: Literal["manual", "ai_saved"] = "manual"
    based_on_catalog_exercise_id: str | None = None
    based_on_modification: str | None = None


class UpdateInstructorExerciseRequest(BaseModel):
    name_he: str | None = Field(default=None, min_length=1)
    name_en: str | None = None
    description_he: str | None = None
    difficulty_level: Literal["beginner", "intermediate", "advanced"] | None = None
    body_focus: list[str] | None = None
    possible_equipment: list[str] | None = None
    teaching_notes: str | None = None
    is_archived: bool | None = None
